### PR TITLE
feat(tls): Add ability to add multiple ca certificates

### LINKED ISSUE #1629: feat: permit multiple ca_certificates
### Motivation

Sometimes there are multiple ca's to be added to the trust store. For instance, when migrating between setups (e.g. going from an internal manual CA to a SPIRE setup). Having only one ca_certificate means a flag-day event is required rather than permitting graceful migration component by component.

### Proposal

Change the config builder to have a list of ca's to add; change the ca_certificate call to appends to the vector.

### PR DESCRIPTION
**Pull request by tottoto**
Resolves #1629.

### CODE REVIEW

**Review by djc — APPROVED** [review]
Is the next step here to get rid of the `Certificate` abstraction around `CertificateDer<'_>`? Would be nice!

**Comment by tottoto** [comment]
I would think that `Certificate` interface cannot be removed to support PEM encoded certificate file.

**Comment by djc** [comment]
Well, the caller would rely on rustls-pemfile so it's one less mandatory dependency to carry within tonic.

**Comment by tottoto** [comment]
Considering `rustls-pemfile` has major version release, I would think it would be acceptable to depend on the crate for the purpose of easiness to use.

**Comment by tottoto** [comment]
Added a method to add multiple ca certificates at once, which is useful when having some sets of certificates.

File: tonic/src/transport/channel/tls.rs [hunk 1]
```diff
@@ -51,6 +51,13 @@ impl ClientTlsConfig {
         ClientTlsConfig { certs, ..self }
     }
 
+    /// Sets the multiple CA Certificates against which to verify the server's TLS certificate.
+    pub fn ca_certificates(self, ca_certificates: Vec<Certificate>) -> Self {
```
Comment (djc) [inline tonic/src/transport/channel/tls.rs:55]: Let's take an `impl IntoIterator<Item = Certificate>` here, instead?

**Review by djc — APPROVED** [review]

File: tonic/src/transport/channel/tls.rs (hunk 1 above)
Comment (tottoto) [reply tonic/src/transport/channel/tls.rs]: Sounds good to me. Updated to use it.

**Pull request merged**